stack_deployment_run.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// StackDeploymentRuns describes all the stack deployment runs-related methods that the HCP Terraform API supports.
+type StackDeploymentRuns interface {
+	// List returns a list of stack deployment runs for a given deployment group.
+	List(ctx context.Context, deploymentGroupID string, options *StackDeploymentRunListOptions) (*StackDeploymentRunList, error)
+}
+
+// stackDeploymentRuns implements StackDeploymentRuns.
+type stackDeploymentRuns struct {
+	client *Client
+}
+
+var _ StackDeploymentRuns = &stackDeploymentRuns{}
+
+// StackDeploymentRun represents a stack deployment run.
+type StackDeploymentRun struct {
+	ID          string    `jsonapi:"primary,stacks-deployment-runs"`
+	Status      string    `jsonapi:"attr,status"`
+	StartedAt   time.Time `jsonapi:"attr,started-at,iso8601"`
+	CompletedAt time.Time `jsonapi:"attr,completed-at,iso8601"`
+
+	// Relationships
+	StackDeploymentGroup *StackDeploymentGroup `jsonapi:"relation,stacks-deployment-groups"`
```

Comment:
The jsonapi struct tag should use `stack-deployment-group` here:

sample response from the API:
```json
{
    "id": "sdr-CmfdnESLTCfzURyn",
    "type": "stacks-deployment-runs",
    "attributes": {
        "status": "pre_deploying",
        "deployment": "single",
        "created-at": "2025-06-16T17:23:01.551Z",
        "updated-at": "2025-06-16T17:23:02.513Z"
    },
    "relationships": {
        "stack-deployment-group": {
            "data": {
                "id": "sdg-voyuv2aFSaZhox5V",
                "type": "stacks-deployment-groups"
            }
        }
    }
}
 ```